galsim/phase_screens.py

```diff
@@ -0,0 +1,623 @@
+                         [Default: 0.03]
+    @param altitude      Altitude of phase screen in km.  This is with respect to the telescope, not
+                         sea-level.  [Default: 0.0]
+    @param L0            Outer scale in meters.  The turbulence power spectrum will smoothly
+                         approach a constant at scales larger than L0.  Set to `None` or `np.inf`
+                         for a power spectrum without an outer scale.  [Default: 25.0]
+    @param speed         Wind speed in meters/second.  [Default: 0.0]
+    @param direction     Wind direction as galsim.Angle [Default: 0.0 * galsim.degrees]
+    @param alpha         Fraction of phase that is "remembered" between time_steps.  The fraction
+                         1-alpha is then the amount of turbulence freshly generated in each step.
+                         [Default: 1.0]
+    @param screen_scale  Physical pixel scale of phase screen in meters.  A fraction of the Fried
+                         parameter is usually sufficiently small, but users should test the effects
+                         of this parameter to ensure robust results.
+                         [Default: same as each screen's r0_500]
+    @param rng           Random number generator as a galsim.BaseDeviate().  If None, then use the
+                         clock time or system entropy to seed a new generator.  [Default: None]
+    """
+    # Fill in screen_size here, since there isn't a default in AtmosphericScreen
+    kwargs['screen_size'] = _listify(screen_size)
+
+    # Set default r0_500 here, so that by default it gets broadcasted below such that the
+    # _total_ r0_500 from _all_ screens is 0.2 m.
+    if 'r0_500' not in kwargs:
+        kwargs['r0_500'] = [0.2]
+    kwargs['r0_500'] = _listify(kwargs['r0_500'])
+
+    # Turn speed, direction into vx, vy
+    if 'speed' in kwargs:
+        kwargs['speed'] = _listify(kwargs['speed'])
+        if 'direction' not in kwargs:
+            kwargs['direction'] = [0*galsim.degrees]*len(kwargs['speed'])
+        kwargs['vx'], kwargs['vy'] = zip(*[v*d.sincos()
+                                           for v, d in zip(kwargs['speed'],
+                                                           kwargs['direction'])])
+        del kwargs['speed']
+        del kwargs['direction']
+
+    # Determine broadcast size
+    nmax = max(len(v) for v in kwargs.values() if hasattr(v, '__len__'))
+
+    # Broadcast r0_500 here, since logical combination of indiv layers' r0s is complex:
+    if len(kwargs['r0_500']) == 1:
+        kwargs['r0_500'] = [nmax**(3./5) * kwargs['r0_500'][0]] * nmax
+
+    if rng is None:
+        rng = galsim.BaseDeviate()
+    kwargs['rng'] = [galsim.BaseDeviate(rng.raw()) for i in xrange(nmax)]
+    return galsim.PhaseScreenList(AtmosphericScreen(**kw) for kw in _lod_to_dol(kwargs, nmax))
+
+
+# Some utilities for working with Zernike polynomials
+# Combinations.  n choose r.
+def _nCr(n, r):
```

Comment:
I find this a bit scary if we're using arbitrary order zernikes because factorials can do bad things when they get huge.  Use the solution here?  http://stackoverflow.com/questions/3025162/statistics-combinations-in-python